storage/indexer/src/db_indexer.rs

```diff
@@ -441,6 +526,60 @@ impl DBIndexer {
         Ok(version)
     }
 
+    pub fn get_state_value_for_resource(
+        &self,
+        address: &AccountAddress,
+        struct_tag: &StructTag,
+    ) -> Result<Option<StateValue>> {
+        let state_view = self
+            .main_db_reader
+            .latest_state_checkpoint_view()
+            .expect("Failed to get state view");
+
+        //let struct_tag = StructTag::from_str(struct_tag_str)?;
+        let state_key = StateKey::resource(address, struct_tag)?;
+        let maybe_state_value = state_view.get_state_value(&state_key)?;
+        Ok(maybe_state_value)
+    }
+
+    pub fn translate_event_v2_to_v1(
+        &self,
+        v2: &ContractEventV2,
+    ) -> Result<Option<ContractEventV1>> {
+        if let Some(translator) = self
+            .event_v2_translation_engine
+            .translators
+            .get(v2.type_tag())
+        {
+            return Ok(Some(translator.translate_event_v2_to_v1(v2, self)?));
+        }
+        Ok(None)
+    }
+
+    fn cache_sequence_number(&self, event_key: &EventKey, sequence_number: u64) {
+        let mut cache = self.event_sequence_number_cache.lock().unwrap();
```

Comment:
could you use `DashMap` instead?

Reply:
@lightmark , `event_sequence_number_cache: Mutex<HashMap<EventKey, u64>>` is implemented using HashMap.

Reply:
I mean, DashMap has a better perf?

Reply:
@lightmark , oh, I thought `DashMap` was a type of `HashMap` :) I will look into `DashMap`.

Reply:
Updated the code with DashMap.